pkg/test/client.go

```diff
@@ -93,8 +93,8 @@ func (f *frameworkClient) Get(gCtx goctx.Context, key dynclient.ObjectKey, obj r
 	return f.Client.Get(gCtx, key, obj)
 }
 
-func (f *frameworkClient) List(gCtx goctx.Context, opts *dynclient.ListOptions, list runtime.Object) error {
-	return f.Client.List(gCtx, opts, list)
+func (f *frameworkClient) List(gCtx goctx.Context, opts dynclient.ListOptionFunc, list runtime.Object) error {
```

Comment:
Should we change the order of the *frameworkClient `List` function parameters to match the upstream change (and so that we can pass multiple `ListOptionFunc`s if necessary.

```go
func (f *frameworkClient) List(
	gCtx goctx.Context,
	list runtime.Object,
	opts ...dynclient.ListOptionFunc,
) error {
	return f.Client.List(gCtx, list, opts...)
}